--- app/api/db/repositories/presenca_repo.py
+++ app/api/db/repositories/presenca_repo.py
@@ -62,3 +62,8 @@
             self.session.commit()
             return True
         return False
+
+    async def deletar_por_usuario(self, usuario_id: int) -> bool:
+        self.session.query(PresencaModel).filter(PresencaModel.usuario_id == usuario_id).delete()
+        self.session.commit()
+        return True
